x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/details/index.tsx

```diff
@@ -39,198 +41,261 @@ import { getStepsData } from '../helpers';
+    const subTitle = useMemo(
+      () =>
+        loading === true || rule === null ? (
+          <EuiLoadingSpinner size="m" />
+        ) : (
+          [
             <FormattedMessage
-              id="xpack.siem.detectionEngine.ruleDetails.ruleUpdateDescription"
-              defaultMessage="Updated by: {by} on {date}"
+              id="xpack.siem.detectionEngine.ruleDetails.ruleCreationDescription"
+              defaultMessage="Created by: {by} on {date}"
               values={{
-                by: rule?.updated_by ?? i18n.UNKNOWN,
+                by: rule?.created_by ?? i18n.UNKNOWN,
                 date: (
                   <FormattedDate
-                    value={rule?.updated_at ?? new Date().toISOString()}
-                    fieldName="updatedAt"
+                    value={rule?.created_at ?? new Date().toISOString()}
+                    fieldName="createdAt"
                   />
                 ),
               }}
-            />
-          ) : (
-            ''
-          ),
-        ]
-      ),
-    [loading, rule]
-  );
-
-  const signalDefaultFilters = useMemo(
-    () => (ruleId != null ? buildSignalsRuleIdFilter(ruleId) : []),
-    [ruleId]
-  );
-  return (
-    <>
-      <WithSource sourceId="default">
-        {({ indicesExist, indexPattern }) => {
-          return indicesExistOrDataTemporarilyUnavailable(indicesExist) ? (
-            <GlobalTime>
-              {({ to, from }) => (
-                <StickyContainer>
-                  <FiltersGlobal>
-                    <SiemSearchBar id="global" indexPattern={indexPattern} />
-                  </FiltersGlobal>
-
-                  <WrapperPage>
-                    <HeaderPage
-                      backOptions={{
-                        href: `#${DETECTION_ENGINE_PAGE_NAME}/rules`,
-                        text: i18n.BACK_TO_RULES,
-                      }}
-                      badgeOptions={{ text: i18n.EXPERIMENTAL }}
-                      border
-                      subtitle={subTitle}
-                      subtitle2={[
-                        lastSignals != null ? (
-                          <>
-                            {detectionI18n.LAST_SIGNAL}
-                            {': '}
-                            {lastSignals}
-                          </>
-                        ) : null,
-                        'Status: Comming Soon',
-                      ]}
-                      title={title}
-                    >
-                      <EuiFlexGroup alignItems="center">
-                        <EuiFlexItem grow={false}>
-                          <RuleSwitch
-                            id={rule?.id ?? '-1'}
-                            enabled={rule?.enabled ?? false}
-                            optionLabel={i18n.ACTIVATE_RULE}
-                          />
-                        </EuiFlexItem>
+            />,
+            rule?.updated_by != null ? (
```

Comment:
This is confusing for a few reasons and the intention is not really there.

The type for `updated_by` and `created_by` both say they cannot be `null` but looking at the operators it does look like they can be which I think is the actual use case. So I would change the types.

The second part is that when `updated_by` is null it will never hit these lines below and show an updated_by by UNKNOWN but rather just show an empty string.

So which is it we want? To show nothing for `updated_by` when it is null or to show it but mark it as `UNKNOWN`?

Reply:
Oooh good catch. This is existing code that got reformatted from my changes, so will have to hear from @XavierM on this, but my best guess is that we'd want the latter and show `UNKNOWN`. Although there may be some special logic I'm missing out on for how `updated_by` is used?